rich/text.py

```diff
@@ -434,11 +435,94 @@ def copy(self) -> "Text":
         copy_self._spans[:] = self._spans
         return copy_self
 
+    def format(self, *args, **kwargs) -> "Text":
+        """Return a new Text instance with formatted specified value(s) and insert them inside the string's placeholder"""
+
+        initial_parsing = [x for x in Formatter().parse(self.plain)]
+        if len(initial_parsing) == 1 and initial_parsing[0][1] == None:
+            return self.copy()
+        formatted = ""
+        parsed_len = 0
+        _args = list(args)
+        _kwargs = kwargs
+        _spans = [[s.start, s.end, s.style] for s in self._spans]
+        if len(_args) != 0 or len(kwargs) != 0:
+            _field_numbers = [
```

Comment:
Not clear what is going on here.

Reply:
converted Span to a list cause can't set attributes to them